app/lib/features/home/views/home_body.dart

```diff
@@ -0,0 +1,528 @@
+                        // adapt layout according to platform.
+                        Breakpoints.small: SlotLayout.from(
+                          key: const Key('primaryNavigation'),
+                          builder: (BuildContext ctx) => SidebarWidget(
+                            labelType: NavigationRailLabelType.selected,
+                            navigationShell: widget.navigationShell,
+                          ),
+                        ),
+                        Breakpoints.mediumAndUp: SlotLayout.from(
+                          key: const Key('primaryNavigation'),
+                          builder: (BuildContext ctx) => SidebarWidget(
+                            labelType: NavigationRailLabelType.all,
+                            navigationShell: widget.navigationShell,
+                          ),
+                        ),
+                      },
+                    )
+                  : null,
+              body: SlotLayout(
+                config: <Breakpoint, SlotLayoutConfig>{
+                  Breakpoints.smallAndUp: SlotLayout.from(
+                    key: const Key('Body Small'),
+                    builder: (BuildContext ctx) => widget.navigationShell,
+                  ),
+                },
+              ),
+              bottomNavigation: !isDesktop &&
+                      keyboardVisibility.valueOrNull != true
+                  ? SlotLayout(
+                      config: <Breakpoint, SlotLayoutConfig>{
+                        //In desktop, we have ability to adjust windows res,
+                        // adjust to navbar as primary to smaller views.
+                        Breakpoints.smallAndUp: SlotLayout.from(
+                          key: Keys.mainNav,
+                          inAnimation: AdaptiveScaffold.bottomToTop,
+                          outAnimation: AdaptiveScaffold.topToBottom,
+                          builder: (BuildContext ctx) => BottomNavigationBar(
+                            showSelectedLabels: false,
+                            showUnselectedLabels: false,
+                            currentIndex: widget.navigationShell.currentIndex,
+                            onTap: (index) {
+                              widget.navigationShell.goBranch(
+                                index,
+                                initialLocation: index ==
+                                    widget.navigationShell.currentIndex,
+                              );
+                            },
+                            items: bottomBarNav,
+                            type: BottomNavigationBarType.fixed,
+                          ),
+                        ),
+                      },
+                    )
+                  : null,
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+
+  void onStateChange(VerificationState? prev, VerificationState next) {
+    if (prev == next) return; // avoid too frequent callbacks
```

Comment:
why would that ever happen? Isn't the better way to test for that before setting the `state`? So that we only ever triggered an update if there really was one?

Reply:
Fixed by https://github.com/acterglobal/a3/pull/1257/commits/517e209e933acad917d0c878c5758f938638146a